sympy/solvers/inequalities.py

```diff
@@ -982,3 +987,457 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+class UnboundedLinearProgrammingError(Exception):
+    """
+    A linear programing problem is said to be unbounded if its objective function can assume arbitrarily large positive
+    values.
+
+    Example
+    =======
+
+    Suppose you want to maximize
+        2x
+    subject to
+        x >= 0
+
+    Clearly there's no upper limit that 2x can take. If you try to use the linear_programing function on such a problem
+    you will get this exception.
+    """
+    pass
+
+
+class InfeasibleLinearProgrammingError(Exception):
+    """
+    A linear programing problem is considered infeasible if its constraint set is empty. That is, if the set of all
+    vectors satisfying the contraints is empty, then the problem is infeasible.
+
+    Example
+    =======
+
+    Suppose you want to maximize
+        x
+    subject to
+        x >= 10
+        x <= 9
+
+    Clearly it's not possible for x to satisfy the given contraints. If you try to use the linear_programing function on
+    a problem where it's not possible to satisfy the contraints, you will get this sort of exception.
+    """
+    pass
+
+
+def _pivot(M, i, j):
+    """
+    There are four rules for the pivot operation:
+    - The pivot quantity goes into its reciprocal
+    - Entries in the same row as the pivot are divided by the pivot
+    - Entries in the same column as the pivot are divided by the pivot and changed in sign.
+    - The remaining entries are reduced in value by the following: the product of the corresponding entries in the same
+    row and column as themselves and the pivot, divided by the pivot.
+
+    Example
+    =======
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.inequalities import _pivot
+    >>> m = Matrix([[3,1,5], \
+                    [2,-3,1], \
+                    [0,3,1]])
+    >>> _pivot(m, 1,0) # pivot around 2, which is in row 1 and col 0
+    Matrix([
+    [-3/2, 11/2, 7/2],
+    [ 1/2, -3/2, 1/2],
+    [   0,    3,   1]])
+    """
+    Mi, Mj, Mij = M[i,:], M[:,j], M[i,j]
+    if Mij == 0:
+        raise ZeroDivisionError("Tried to pivot about zero-valued entry.")
+    MM = M - Mj * (Mi / Mij)
+    MM[i,:] = Mi / Mij
+    MM[:,j] = -Mj / Mij
+    MM[i,j] = 1 / Mij
+    return MM
+
+
+def _simplex(M, skip_phase_2=False):
+    """
+    Simplex method with randomized pivoting
+
+    LINEAR PROGRAMMING: A Concise Introduction by Thomas S. Ferguson
+      - http://web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_LP.pdf
```

Comment:
This should be in a references section:
https://docs.sympy.org/latest/contributing/docstring.html#references-section